addons/pos_hr/static/src/js/models.js

```diff
@@ -81,8 +81,8 @@ models.Order = models.Order.extend({
     },
     init_from_JSON: function (json) {
         super_order_model.init_from_JSON.apply(this, arguments);
-        if (this.pos.config.module_pos_hr) {
-            this.employee = this.pos.employee_by_id[json.employee_id];
+        if (this.pos.config.module_pos_hr && json.employee_id) {
+            this.employee = this.pos.employee_by_id[json.employee_id[0]];
```

Comment:
`json.employee_id` is an id and not a `[id, name]`, so this is assigning to undefined in normal point_of_sale

Reply:
@trj-odoo maybe I'm missing something but I think that json.employee_id is `[id,name]` . https://i.imgur.com/mkGsfjw.png

Reply:
Yes only in `pos_restaurant` because you're doing `search_read`. But in `pos_hr` only, the `json.employee_id` is a simple id. You have to know that export_JSON and init_JSON are not only used to send data to the back end but also to store data in the localStorage of the browser. So when you create an order and then you refresh the page (again, in normal point_of_sale), the cashier of the order become undefined because `json.employee_id` is a simple `number`.
What you can do is either continue with what you're doing and make the `json.employee_id` a `[id, name]` in `pos_hr` or going along with the `json.employee_id = number` then you need to override `get_table_draft_orders` from `pos_restaurant` in `pos_hr` (then you need to create the method in `point_of_sale` which means that we do really need a `pos_restaurant_hr` module) and overwrite for each order: `order['employee_id'] = order['employee_id'][0]`

Reply:
see
```js
    export_as_JSON: function () {
        const json = super_order_model.export_as_JSON.apply(this, arguments);
        if (this.pos.config.module_pos_hr) {
            json.employee_id = this.employee ? this.employee.id : false;
        }
        return json;
    },
```